lib/client/doc.js

```diff
@@ -571,15 +576,15 @@ Doc.prototype._otApply = function(op, source) {
 
     // The 'before op' event enables clients to pull any necessary data out of
     // the snapshot before it gets changed
-    this.emit('before op', op.op, source);
+    this.emit('before op', op.op, source, op.src);
```

Comment:
Let's think about if we can get rid of this extra `src` emission. We basically need to know if the op is our own op in `RemoteDocPresence._handleOp`, and in `Backend.transformPresenceToLatestVersion`

Reply:
Let's keep this, but bury it one level deep in an object with a `src` property for forwards-compatibility and self-documentation. We should also update the `README`.